dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/Admin/Users.cshtml.cs

```diff
@@ -117,4 +122,57 @@ public record UserInfo
         public required string? Trn { get; init; }
         public required TrnLookupStatus? TrnLookupStatus { get; init; }
     }
+
+    private async Task<Expression<Func<User, bool>>> GetFilterPredicate(TrnLookupStatus?[] lookupStatus, string? userSearch)
+    {
+        var filterPredicate = PredicateBuilder.New<User>(user => user.UserType == UserType.Default);
+
+        if (lookupStatus.Length > 0)
+        {
+            filterPredicate.And(user => lookupStatus.Contains(user.TrnLookupStatus));
+        }
+
+        if (!string.IsNullOrEmpty(userSearch))
+        {
+            switch (userSearch)
+            {
+                case var searchString when new Regex(@"^\d{7}$").IsMatch(searchString):
+                    filterPredicate.And(user => user.Trn == searchString);
+                    break;
+                case var searchString when new Regex(@".*@.*").IsMatch(searchString):
+                    filterPredicate.And(user => user.EmailAddress == searchString);
+                    break;
+                default:
+                    var userIds = await GetUsersByName(userSearch);
+                    filterPredicate.And(user => userIds.Contains(user.UserId));
+                    break;
+            }
+        }
+
+        return filterPredicate;
+    }
+
+    private async Task<Guid[]> GetUsersByName(string searchString)
+    {
+        var names = searchString.Split(' ').Take(2).ToArray();
+
+        var searchPredicate = PredicateBuilder.New<UserSearchAttribute>(
+            a => a.AttributeType == "first_name" && a.AttributeValue == names[0]);
```

Comment:
Searching for exact match with names. Can do a contains, but may need to consider ranking the matches